.github/workflows/ci.yml

```diff
@@ -0,0 +1,124 @@
+name: Run Tests
+
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+
+env:
+  PKG_DEPS_UBUNTU: >-
+    libdb-dev
+
+jobs:
+  dist:
+    name: Make distribution using Dist::Zilla
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Cache ~/perl5
+        uses: actions/cache@v3
+        with:
+          key: ${{ runner.os }}-dist-locallib
+          path: ~/perl5
+      - name: Perl version
+        run: |
+          perl -v
+      - name: Install cpanm
+        run: |
+          curl -L https://cpanmin.us | perl - --sudo App::cpanminus
+      - name: Install local::lib
+        run: |
+          cpanm --local-lib=~/perl5 local::lib && eval $(perl -I ~/perl5/lib/perl5/ -Mlocal::lib)
+      - name: Install Dist::Zilla
+        shell: bash
+        run: |
+          eval $(perl -I ~/perl5/lib/perl5/ -Mlocal::lib)
+          cpanm -n Dist::Zilla
+          dzil authordeps --missing | cpanm -n
+      - name: Make distribution
+        shell: bash
+        run: |
+          eval $(perl -I ~/perl5/lib/perl5/ -Mlocal::lib)
+          dzil build --in build-dir
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: ./build-dir
+  test:
+    needs: dist
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-latest]
+        perl: ['5']
+        author-testing: [false]
+        include:
+          - { os: 'ubuntu-latest', perl: "5"    , author-testing: true  , coverage: true }
+          - { os: 'ubuntu-latest', perl: "5.16" }
+          - { os: 'ubuntu-latest', perl: "5.20" }
+          - { os: 'ubuntu-latest', perl: "5.30" }
+          - { os: 'ubuntu-latest', perl: "5.32" }
+          - { os: 'ubuntu-latest', perl: "5.36" }
```

Comment:
Can you say what motivates the selection of these versions, and needing to have 5 explicitly specified ones?

Reply:
I believe I was looking at CPAN Testers to find a minimum version that had no failures and then added a couple more versions spaced apart to get a  sample (this could be changed to test against every stable version). While Perl is very backwards compatible, sometimes getting an idea of when something like a warning appears can help narrow down where it is coming from.

It should probably add Perl 5.6 as well since that is what the dist explicitly has as  the minimum version, but this looks wrong after I scanned the code to find if it used newer features:

```shell
$ perlver lib/ > output.log

$ grep -vP 'v5\.[56]\.' output.log  | grep 'v5' | cut -d'|' -f2 | head -n -2  | tr -d ' ' | xargs -n1 perlver --blame

 ------------------------------------------------------------
 File    : lib/BioPerl.pm
 Line    : 3
 Char    : 1
 Rule    : _pragma_utf8
 Version : 5.008
 ------------------------------------------------------------
 use utf8;
 ------------------------------------------------------------

 ------------------------------------------------------------
 File    : lib/Bio/PullParserI.pm
 Line    : 296
 Char    : 21
 Rule    : _open_scalar
 Version : 5.008
 ------------------------------------------------------------
 open my $fake_fh, "+<", \$thing or $self->throw("Could not open file '$thing': $!");
 ------------------------------------------------------------

 ------------------------------------------------------------
 File    : lib/Bio/Root/IO.pm
 Line    : 288
 Char    : 13
 Rule    : _binmode_2_arg
 Version : 5.008
 ------------------------------------------------------------
 binmode
 ------------------------------------------------------------

 ------------------------------------------------------------
 File    : lib/Bio/Tree/Tree.pm
 Line    : 393
 Char    : 5
 Rule    : _open_scalar
 Version : 5.008
 ------------------------------------------------------------
 open my $fh, '>', \$string or $self->throw("Could not write '$string' as file: $!");
 ------------------------------------------------------------

 ------------------------------------------------------------
 File    : lib/Bio/Tools/GuessSeqFormat.pm
 Line    : 465
 Char    : 9
 Rule    : _open_scalar
 Version : 5.008
 ------------------------------------------------------------
 open $fh, '<', \$text or $self->throw("Could not read from string: $!");
 ------------------------------------------------------------
```

So the minimum needs to be bumped to Perl v5.8. I'll open an issue for that.

Having the minimum version would help with finding issues where a change actually requires a newer Perl version.

Reply:
Issue at <https://github.com/bioperl/bioperl-live/issues/375>.